lib/rmt/downloader.rb

```diff
@@ -27,24 +27,25 @@ def initialize(repository_url:, local_path:, auth_token: nil, logger: nil, cache
   def download(remote_file, checksum_type: nil, checksum_value: nil, use_cache: false)
     local_file = self.class.make_local_path(@local_path, remote_file)
 
-    if_modified_since = nil
+    cache_timestamp = nil
     if use_cache
       raise 'Cache path not set!' unless @cache_path
       cache_file = File.join(@cache_path, remote_file)
-      if_modified_since = File.mtime(cache_file).utc.rfc2822 if File.exist?(cache_file)
+      cache_timestamp = File.mtime(cache_file).utc.rfc2822 if File.exist?(cache_file)
```

Comment:
Thanks! Reads so much better now! :+1:

Reply:
I found out why it does not work for me. Your time format resultet in "Thu, 25 Jan 2018 09:34:05 -0000" on my machine.
When I replace that with `File.mtime(cache_file).gmtime.strftime('%a, %d %b %Y %T GMT')` it works.
I haven't found a method or %-shortcut for that. :confused:

Reply:
[There's also](https://ruby-doc.org/stdlib-2.2.2/libdoc/time/rdoc/Time.html#method-i-httpdate) `httpdate` method, it produces the date in the correct format.